### PR TITLE
chore(eslint): ESLint JSDoc rules, JSDoc improvements

### PR DESCRIPTION
**Pull request by hekike**
Add ESLint rules for JSDoc and improve JSDoc.

# Changes

- Add ESLint rules for JSDoc
- Improve JSDoc

### CODE REVIEW

**Comment by DonutEspresso** [comment]
This looks great! 🎉    I'm not a fan of the hyphen in the jsdoc, but don't feel super strongly about it.

File: docs/_api/plugins.md [hunk 1]
```diff
@@ -745,54 +753,55 @@ Returns **[Function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Ref
 
 ### cpuUsageThrottle
 
-cpuUsageThrottle is a middleware that rejects a variable number of requests
-(between 0% and 100%) based on a historical view of CPU utilization of a
-Node.js process. Essentially, this plugin allows you to define what
-constitutes a saturated Node.js process via CPU utilization and it will
-handle dropping a % of requests based on that definiton. This is useful when
-you would like to keep CPU bound tasks from piling up causing an increased
+cpuUsageThrottle is a middleware that rejects a variable number of requests
```
Comment (yunong) [inline docs/_api/plugins.md]: Trailing spaces? I think your editor may not bet stripping these?

**Comment by hekike** [comment]
@yunong right, I fixed my editor and removed all the trailing whitespaces.

File: docs/_api/plugins.md (hunk 1 above)
Comment (hekike) [reply docs/_api/plugins.md]: `docs/_api/*.md` is auto-generated, I fixed whitespaces at other places.

**Pull request merged**